### DISCUSSION TITLE
Fade animation when switching between data

### DISCUSSION THREAD

**Discussion by joseandroidengineer** [patrykandpatrick/vico#844] [Questions] [answered]
### Question

Hello! Currently when switching between data series the charts animate from the bottom x-axis label to the proper graph data set.
I was wondering if there is a way to animate where it fades to another set of coordinates instead of starting with 0.

Let me know if you need more information. Thank you so much

### Vico version(s)

2.0.0 alpha.21

### UI framework(s)

Jetpack Compose

**Comment by Gowsky** [comment, marked as answer]
Hello, @joseandroidengineer.

> Currently when switching between data series the charts animate from the bottom x-axis label to the proper graph data set.

Please note that this happens only for new _x_ values. When both the old series and the new series have a _y_ value for a given _x_ value, the _y_ value is animated from the old one to the new one.

> I was wondering if there is a way to animate where it fades to another set of coordinates instead of starting with 0.

Yes, you can do this by creating a custom `DrawingModelInterpolator` (`CartesianLayerDrawingModelInterpolator` in newer versions).
